src/lein_monolith/task/fingerprint.clj

```diff
@@ -0,0 +1,352 @@
+(ns lein-monolith.task.fingerprint
+  (:require
+    [clojure.data]
+    [clojure.edn :as edn]
+    [clojure.java.io :as jio]
+    [clojure.set :as set]
+    [clojure.string :as str]
+    [leiningen.core.main :as lein]
+    [leiningen.core.project :as project]
+    [lein-monolith.dependency :as dep]
+    [lein-monolith.target :as target]
+    [lein-monolith.task.util :as u]
+    [multihash.core :as mhash]
+    [multihash.digest :as digest]
+    [puget.color.ansi :as ansi]
+    [puget.printer :as puget])
+  (:import
+    (java.io
+      File
+      PushbackInputStream)))
+
+
+;; ## Options
+
+(def selection-opts
+  (assoc target/selection-opts :upstream 0 :downstream 0))
+
+
+;; ## Hashing projects' inputs
+
+(def ^:private ->multihash
+  "Globally storing the algorithm we use to generate each multihash."
+  digest/sha1)
+
+
+(defn- aggregate-hashes
+  "Takes a collection of multihashes, and aggregates them together into a unified hash."
+  [mhashes]
+  ;; TODO: is there a better way to do this?
+  (if (= 1 (count mhashes))
+    (first mhashes)
+    (->> mhashes
+         (map mhash/base58)
+         (sort)
+         (apply str)
+         (->multihash))))
+
+
+(defn- list-all-files
+  [^File file]
+  (if (.isFile file)
```

Comment:
Do you want to recurse into symlinked directories? If not, you should invert this conditional and check `isDirectory` before listing.

Reply:
I hadn't thought about this, but I think I do want to follow symlinks. That way if someone decides to have a symlink in `src` for some reason, we'll treat those files the same. I haven't tested whether this works with checkouts or not though.